crates/solver/src/settlement/settlement_encoder.rs

```diff
@@ -687,6 +687,8 @@ pub mod tests {
         assert!(settlement
             .add_liquidity_order_trade(order01.clone(), 20.into(), 0.into())
             .is_ok());
+        // ensures that the output of add_liquidity_order is sorted
+        assert_eq!(settlement.tokens, vec![token(0), token(1)]);
```

Comment:
This still leaves room for forgetting to update the indices after adding a new token. The following assertions would have caught that.
```rust
assert_eq!(
    settlement.liquidity_order_trades[0].trade.sell_token_index, // <-- is the sell token index of liquidity order
    0
);
```

Reply:
Added in here: 
[50ecc3d](https://github.com/gnosis/gp-v2-services/pull/1544/commits/50ecc3d7230bcd04df1c2120f64f29e5c1cc6e6f)

But I don't think that it is important, as updating and sorting is always done "atomic" in the function sort_and_update_indicies. But yeah, now we are safe. thanks

Reply:
I think it's nice to have that extra peace of mind. Thanks for adding that. 👍

Reply:
I like the added assert as it protects us from refactoring errors that may happen in the future (if we decide to change how `sort_and_update_indicies` works for example).